data/publisher/src/main/java/tech/pegasys/teku/data/publisher/BeaconNodeMetricData.java

```diff
@@ -13,64 +13,97 @@
 
-import com.fasterxml.jackson.annotation.JsonCreator;
-import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import java.util.Objects;
+import java.io.File;
+import org.apache.commons.io.FileUtils;
+import tech.pegasys.teku.infrastructure.metrics.MetricsPublishCategories;
 
-@JsonInclude(JsonInclude.Include.NON_NULL)
-public class BeaconNodeMetricData extends BaseMetricData {
+public class BeaconNodeMetricData extends GeneralProcessMetricData {
+  @JsonProperty("disk_beaconchain_bytes_total")
+  private final long diskBytesTotal;
 
-  public final Long cpu_process_seconds_total;
-  public final Long memory_process_bytes;
-  public final String client_name;
-  public final String client_version;
-  public final Integer network_peers_connected;
-  public final Long sync_beacon_head_slot;
+  @JsonProperty("network_libp2p_bytes_total_receive")
+  private final long libp2pBytesReceived;
+
+  @JsonProperty("network_libp2p_bytes_total_transmit")
+  private final long libp2pBytesTotalTransmitted;
+
+  @JsonProperty("network_peers_connected")
+  private final int networkPeersConnected;
+
+  @JsonProperty("sync_eth1_connected")
+  private final boolean eth1Connected;
+
+  @JsonProperty("sync_eth2_synced")
+  private final boolean eth2Synced;
+
+  @JsonProperty("sync_beacon_head_slot")
+  private final long headSlot;
+
+  @JsonProperty("sync_eth1_fallback_configured")
+  private final boolean eth1FallbackConfigured;
+
+  @JsonProperty("sync_eth1_fallback_connected")
+  private final boolean eth1FallbackConnected;
+
+  @JsonProperty("slasher_active")
+  private final boolean slasherActive = false;
 
-  @JsonCreator
   public BeaconNodeMetricData(
-      @JsonProperty("version") int version,
-      @JsonProperty("timestamp") long timestamp,
-      @JsonProperty("process") String process,
-      @JsonProperty("cpu_process_seconds_total") Long cpuProcessSecondsTotal,
-      @JsonProperty("memory_process_bytes") Long memoryProcessBytes,
-      @JsonProperty("client_name") String clientName,
-      @JsonProperty("client_version") String clientVersion,
-      @JsonProperty("network_peers_connected") Integer networkPeersConnected,
-      @JsonProperty("sync_beacon_head_slot") Long syncBeaconHeadSlot) {
-    super(timestamp, process);
-    this.cpu_process_seconds_total = cpuProcessSecondsTotal;
-    this.memory_process_bytes = memoryProcessBytes;
-    this.client_name = clientName;
-    this.client_version = clientVersion;
-    this.network_peers_connected = networkPeersConnected;
-    this.sync_beacon_head_slot = syncBeaconHeadSlot;
+      final long timestamp, final MetricsPublisherSource source, final File dataDirectory) {
+    super(timestamp, MetricsPublishCategories.BEACON_NODE.getDisplayName(), source);
+    if (dataDirectory != null && dataDirectory.exists() && dataDirectory.isDirectory()) {
+      this.diskBytesTotal = FileUtils.sizeOfDirectory(dataDirectory);
```

Comment:
I worry that this is potentially quite expensive as it has to walk through the entire file tree adding up file sizes. There's just over 15,000 files in the database for our prater validator and nearly 35,000 files for the prater archive node.

I wonder if we need to cache this value and only update at a slower rate since disk usage doesn't change too fast.

Reply:
yeh the metrics call for several of these, we might need a sanity check on the interval... if its every 30 seconds it might be ok, but if its too low then this will all become very expensive...

Reply:
So `FileUtils.sizeOfDirectory` is more efficient by the looks than other things ,but with a decent number of files it really does grind to a hat. with 30k extra files, it took over 1s to compute the storage space.

We could potentially return 0 for now, or rate limit this request to be infrequent if it's taking a while - something like a service that will let the request run if it is performing well and returns a cached result if it was too slow?

For now i think 0 for this PR.

Reply:
Agreed 0 for this PR.  I don't think I'd try to do anything fancy with caching based on performance - I'd just set it to only update every x minutes and use the cached value between those times.  I suspect x is at least 5 minutes but I'd be tempted to make it more like 15.